src/main/java/org/apache/maven/plugin/resources/remote/AbstractProcessRemoteResourcesMojo.java

```diff
@@ -906,46 +932,53 @@ protected void processResourceBundles(ClassLoader classLoader, VelocityContext c
+                    try (CachingOutputStream os = new CachingOutputStream(outputFile)) {
+                        String bundleEncoding = bundle.getSourceEncoding();
+                        if (bundleEncoding == null) {
+                            bundleEncoding = encoding;
                         }
-                    } else {
-                        URL resUrl = classLoader.getResource(bundleResource);
-                        if (resUrl != null) {
-                            FileUtils.copyURLToFile(resUrl, f);
+                        try (Writer writer = new OutputStreamWriter(os, bundleEncoding)) {
+                            velocity.mergeTemplate(bundleResource, bundleEncoding, context, writer);
                         }
                     }
+                } else {
+                    URL resUrl = classLoader.getResource(bundleResource);
+                    if (resUrl != null) {
+                        FileUtils.copyURLToFile(resUrl, outputFile);
+                    }
+                }
 
-                    File appendedResourceFile = new File(appendedResourcesDirectory, projectResource);
-                    File appendedVmResourceFile = new File(appendedResourcesDirectory, projectResource + ".vm");
+                File appendedResourceFile = new File(appendedResourcesDirectory, projectResource);
+                File appendedVmResourceFile = new File(appendedResourcesDirectory, projectResource + ".vm");
 
-                    if (appendedResourceFile.exists()) {
-                        getLog().info("Copying appended resource: " + projectResource);
-                        try (InputStream in = Files.newInputStream(appendedResourceFile.toPath());
-                                OutputStream out = new FileOutputStream(f, true)) {
-                            IOUtil.copy(in, out);
-                        }
+                if (appendedResourceFile.exists()) {
+                    getLog().info("Copying appended resource: " + projectResource);
+                    try (InputStream in = Files.newInputStream(appendedResourceFile.toPath());
+                            OutputStream out = new FileOutputStream(outputFile, true)) {
+                        IOUtil.copy(in, out);
+                    }
 
-                    } else if (appendedVmResourceFile.exists()) {
-                        getLog().info("Filtering appended resource: " + projectResource + ".vm");
+                } else if (appendedVmResourceFile.exists()) {
+                    getLog().info("Filtering appended resource: " + projectResource + ".vm");
 
-                        try (Reader reader = new FileReader(appendedVmResourceFile);
-                                Writer writer = getWriter(bundle, f)) {
-                            Velocity.init();
-                            Velocity.evaluate(context, writer, "remote-resources", reader);
-                        }
+                    try (Reader reader = new FileReader(appendedVmResourceFile);
```

Comment:
needs an encoding, or use Files.newReader instead

Reply:
done and fix encoding in other places